kobo-uncaged/device/device.go

```diff
@@ -353,67 +425,86 @@ func (k *Kobo) readMDfile() error {
+		k.BooksInDB[cid] = true
 	}
 	// Hopefully, our metadata is now up to date. Update the cache on disk
-	err = k.WriteMDfile()
-	if err != nil {
-		return errors.Wrap(err, "error writing metadata to disk")
+	if err = k.WriteMDfile(); err != nil {
+		return fmt.Errorf("readMDfile: error writing metadata to disk: %w", err)
 	}
 	return nil
 }
 
 // WriteMDfile writes metadata to file
 func (k *Kobo) WriteMDfile() error {
 	var n int
-	metadata := make([]KoboMetadata, len(k.MetadataMap))
+	var err error
+	metadata := make([]uc.CalibreBookMeta, len(k.MetadataMap))
 	for _, md := range k.MetadataMap {
 		metadata[n] = md
 		n++
 	}
-	return util.WriteJSON(filepath.Join(k.BKRootDir, calibreMDfile), metadata)
+	if err = util.WriteJSON(filepath.Join(k.BKRootDir, calibreMDfile), metadata); err != nil {
+		err = fmt.Errorf("WriteMDfile: %w", err)
+	}
+	return err
 }
 
 func (k *Kobo) readUpdateMDfile() error {
 	emptyOrNotExist, err := util.ReadJSON(filepath.Join(k.BKRootDir, kuUpdatedMDfile), &k.UpdatedMetadata)
 	if emptyOrNotExist {
 		// ignore
 	} else if err != nil {
-		return errors.Wrap(err, "error reading update metadata JSON")
+		return fmt.Errorf("readUpdateMDfile: error reading update metadata JSON: %w", err)
 	}
 	return nil
 }
 
 // WriteUpdateMDfile writes updated metadata to file
 func (k *Kobo) WriteUpdateMDfile() error {
+	var err error
 	// We only write the file if there is new or updated metadata to write
 	if len(k.UpdatedMetadata) == 0 {
 		return nil
 	}
-	return util.WriteJSON(filepath.Join(k.BKRootDir, kuUpdatedMDfile), k.UpdatedMetadata)
+	// Don't write the file if we are updating metadata via DB trigger
+	if k.KuConfig.AddMetadataByTrigger {
+		return nil
+	}
+	if err = util.WriteJSON(filepath.Join(k.BKRootDir, kuUpdatedMDfile), k.UpdatedMetadata); err != nil {
+		err = fmt.Errorf("WriteUpdateMDfile: error writing update metadata JSON: %w", err)
+	}
+	return err
 }
 
 func (k *Kobo) loadDeviceInfo() error {
 	emptyOrNotExist, err := util.ReadJSON(filepath.Join(k.BKRootDir, calibreDIfile), &k.DriveInfo.DevInfo)
 	if emptyOrNotExist {
 		uuid4, _ := uuid.NewV4()
 		k.DriveInfo.DevInfo.LocationCode = "main"
-		k.DriveInfo.DevInfo.DeviceName = "Kobo " + k.Device.Model()
+		k.DriveInfo.DevInfo.DeviceName = k.Device.Family()
 		k.DriveInfo.DevInfo.DeviceStoreUUID = uuid4.String()
 		if k.useSDCard {
 			k.DriveInfo.DevInfo.LocationCode = "A"
 		}
 	} else if err != nil {
-		return errors.Wrap(err, "error reading device info JSON")
+		return fmt.Errorf("loadDeviceInfo: error reading device info JSON: %w", err)
 	}
 	return nil
 }
 
 // SaveDeviceInfo save device info to file
 func (k *Kobo) SaveDeviceInfo() error {
-	return util.WriteJSON(filepath.Join(k.BKRootDir, calibreDIfile), k.DriveInfo.DevInfo)
+	var err error
+	if err = util.WriteJSON(filepath.Join(k.BKRootDir, calibreDIfile), k.DriveInfo.DevInfo); err != nil {
+		err = fmt.Errorf("SaveDeviceInfo: error saving device info JSON: %w", err)
+	}
+	return err
```

Comment:
That separate err var isn't needed for this. Use this:

```go
	if err := util.WriteJSON(filepath.Join(k.BKRootDir, calibreDIfile), k.DriveInfo.DevInfo); err != nil {
		return fmt.Errorf("SaveDeviceInfo: error saving device info JSON: %w", err)
	}
	return nil
```

Reply:
Thanks, I've been trying to make that change where I spot them